nailgun/entities.py

```diff
@@ -2289,6 +2289,33 @@ def add_puppetclass(self, synchronous=True, **kwargs):
         response = client.post(self.path('puppetclass_ids'), **kwargs)
         return _handle_response(response, self._server_config, synchronous)
 
+    def delete_puppetclass(self, synchronous=True, **kwargs):
+        """Remove a Puppet class from host group
+
+        Here is an example of how to use this method::
+            hostgroup.delete_puppetclass(data={'puppetclass_id': puppet.id})
+
+        Constructs path:
+            /api/hostgroups/:hostgroup_id/puppetclass_ids/:id
+
+        :param synchronous: What should happen if the server returns an HTTP
+            202 (accepted) status code? Wait for the task to complete if
+            ``True``. Immediately return the server's response otherwise.
+        :param kwargs: Arguments to pass to requests.
+        :returns: The server's response, with all JSON decoded.
+        :raises: ``requests.exceptions.HTTPError`` If the server responds with
+            an HTTP 4XX or 5XX message.
+
+        """
+        kwargs = kwargs.copy()
+        kwargs.update(self._server_config.get_client_kwargs())
+        path = "{0}/{1}".format(
+            self.path('puppetclass_ids'),
+            kwargs['data'].pop('puppetclass_id')
```

Comment:
:+1:

A heads up to the other reviewers: the call to `pop()` here is intentional. Satellite expects a puppetclass ID as part of the request path, but not as part of the request body. Another approach would be to change the signature of this method to `delete_puppetclass(self, puppetclass_id, synchronous=True, *kwargs)`. That would work, but it would make this method different from nearly all the other custom methods, which accept arguments of `(self, synchronous=True, **kwargs)`.